### PR TITLE
Gamemode inventories

### PR DESCRIPTION
**Pull request by hupster**
This adds separate inventories for the different gamemodes that can be enforce in the different dimensions. That is the one thing I was missing in this nice mod, for the rest it works for me as the first functional replacement for the old MultiVerse, which supported that. I'm using simpleportals for the portal part.

Details: When a user crosses to a dimensions with a different enforced gamemode, the old inventory is stored to disk and the last used inventory for that gamemode is given to the user. All is stored in a single file per user in NBT format, which is only accessed on dimension changes. Nothing happens when the user logs in or out, storing the active inventory when a user logs in or out is left unchanged. 

Extra features:
* config option
* support for baubles, and easy to add support for more inventory add-ons
* resolve an issue where enforced gamemode is not applied at user login
* make the dynmap mod load later, so all loaded dimensions show on it

### CODE REVIEW

File: src/main/resources/mcmod.info [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "credits": "",
   "logoFile": "",
   "screenshots": [],
-  "dependencies": []
+  "dependencies": [],
+  "dependants": [Dynmap]
```
Comment (maruohon) [inline src/main/resources/mcmod.info]: Shouldn't this be double quoted? And all modids these days are supposed to be all lower case, I'm assuming this is an array of modids?

**Comment by maruohon** [comment]
Thanks! There was actually a mention of this feature on my Discord server just yesterday... coincidences are funny :)

I'll take a closer look at this and pull it soon, after I have the chance to test it myself.

But would you mind fixing the coding style and whitespace usage to match the project's style? Basically these things caught my eye about the aesthetics:
* I use only spaces for indentation (this PR has mixed tabs and spaces in many places)
* I usually use curly braces even on one-line ifs
* I usually leave a blank line before if, for, while
* I use a verbose/"more visible" `== false` instead of `!` for false boolean checks (I know most other people hate this but... oh well)

Or I can also fix the aesthetics myself since I know what type of code style I like looking at exactly, hehe :P

**Comment by hupster** [comment]
No problem :)
You're probably right about the dependants having to be lowercase, I didn't notice because it worked. Actually there might be a better place to resolve to mod loading order without having to mention specific mods, I'm not too familiar with that.

Yes feel free to change the coding style, I could try but you probably know exactly how you want it. If you send a request I'll merge.

What do you think about the idea of an inventory per gamemode? I was in doubt if an inventory per dimension would be better, but I chose this because you probably would want to go to the nether with you survival inventory, for instance. It wouldn't be hard to change the code for per dimension, but you'd need configuration options for sharing, etc.

Meanwhile I'm looking at the many player.dat files that I have in my old bukkit multiverse worlds. Maybe I can make a way to import the inventory if the data is missing in the new file.

**Comment by maruohon** [comment]
It was actually per-dimension inventories that was briefly mentioned on Discord. So yeah, I'd like that to be included too, but I think per-gamemode inventories can also be very useful. So I think the best way would be to have them both as config options. Either two options, the perGamemodeInventories that you added and then another, perDimensionInventories that would simply take precedence when enabled, or a single option where you select none, or the type you want to use.

Also... for even more flexibility, maybe the per-dimension variant should be further expanded and hooked into the JED config system, so that it would be possible for only specific dimensions to have separate inventories. So you could for example have a creative test dimension type thing (although creative mode separate inventory would be handled by the per-gamemode variant... but let's assume there is some other specific dimension where you'd want the player to have a separate inventory).

If you are ok with it, I can take over from here and add the per-dimension inventories and hook it up to the JED configs etc.

**Comment by hupster** [comment]
Sure it's all yours!
I'd be glad to see that all this functionality will be part of future releases.

**Pull request merged**